bin/analysis.cpp

```diff
@@ -237,6 +236,20 @@ int main(int ac, char* av[])
 		return 1;
 	}
 
+
+	std::cout << "[INFO] Selected modes: \n";
+	unsigned int msize = modes.size();
+	modes.erase( std::remove( modes.begin(), modes.end(), "signal" ), modes.end() );
+	if(modes.size() != msize){
```

Comment:
This is to keep all the signal samples separate. If the modes vector is expanded before the getChainNames call, it is easier to process each sample at a time.